libtransmission/tr-lpd.h

```diff
@@ -8,9 +8,47 @@
 #error only libtransmission should #include this header.
 #endif
 
-int tr_lpdInit(tr_session*, tr_address*);
-void tr_lpdUninit(tr_session*);
-bool tr_lpdSendAnnounce(tr_torrent const*);
+#include <ctime>
+#include <memory>
+#include <string_view>
+#include <vector>
 
-/**
-* @} */
+#include "transmission.h"
+
+#include "net.h" // for tr_address, tr_port
+#include "timer.h"
+
+class tr_torrents;
+struct tr_session;
+struct event_base;
+
+class tr_lpd
+{
+public:
+    class Mediator
+    {
+    public:
+        virtual ~Mediator() = default;
+
+        [[nodiscard]] virtual tr_port port() const = 0;
+
+        [[nodiscard]] virtual bool allowsLPD() const = 0;
+
+        struct TorrentInfo
```

Comment:
nit: Maybe struct should go after dtor? or even at the top of definition.
It's just feels out of place, imo.

Reply:
Agreed. My rationale when I wrote it was to declare it near where it was used (the function after it returns a vector of them) but better to use [conventional class member declaration order]( https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-order) :+1:

Moved in 632ca0f227fe3e9771ee7cb984eed54e897d801c